swarm-cluster-using-docker-extension/azuredeploy.json

```diff
@@ -0,0 +1,396 @@
+				"adminUsername": "[parameters('adminUsername')]",
+				"adminPassword": "[parameters('adminPassword')]"
+			},
+			"storageProfile": {
+				"imageReference": {
+					"publisher": "[variables('imagePublisher')]",
+					"offer": "[variables('imageOffer')]",
+					"sku" : "[parameters('ubuntuOSVersion')]",
+					"version":"latest"
+				},
+				"osDisk" : {
+					"name": "osdisk",
+					"vhd": {
+						"uri": "[concat('http://', parameters('newStorageAccountName'), '.blob.core.windows.net/', variables('vmStorageAccountContainerName'), '/', variables('OSDiskName'), 'master-osdisk.vhd')]"
+					},
+					"caching": "ReadWrite",
+					"createOption": "FromImage"
+				}
+			},
+			"networkProfile": {
+				"networkInterfaces": [
+					{
+						"id": "[resourceId('Microsoft.Network/networkInterfaces',concat(variables('nicName'), 'master'))]"
+					}
+				]
+			}
+		}
+	},
+	{
+		"apiVersion": "2015-05-01-preview",
+		"type": "Microsoft.Compute/virtualMachines",
+		"name": "[concat(variables('vmName'), copyIndex())]",
+		"location": "[variables('location')]",
+		"copy": {
+			"name": "vmLoop",
+			"count": "[parameters('slaveCount')]"
+		},
+		"dependsOn": [
+			"[concat('Microsoft.Storage/storageAccounts/', parameters('newStorageAccountName'))]",
+			"[concat('Microsoft.Network/networkInterfaces/', variables('nicName'), copyIndex())]"
+		],
+		"properties": {
+			"hardwareProfile": {
+				"vmSize": "[variables('vmSize')]"
+			},
+			"osProfile": {
+				"computername": "[concat(variables('vmName'), copyIndex())]",
+				"adminUsername": "[parameters('adminUsername')]",
+				"adminPassword": "[parameters('adminPassword')]"
+			},
+			"storageProfile": {
+				"imageReference": {
+					"publisher": "[variables('imagePublisher')]",
+					"offer": "[variables('imageOffer')]",
+					"sku" : "[parameters('ubuntuOSVersion')]",
+					"version":"latest"
+				},
+				"osDisk" : {
+					"name": "osdisk",
+					"vhd": {
+						"uri": "[concat('http://', parameters('newStorageAccountName'), '.blob.core.windows.net/', variables('vmStorageAccountContainerName'), '/', variables('OSDiskName'), copyIndex(), '-osdisk.vhd')]"
+					},
+					"caching": "ReadWrite",
+					"createOption": "FromImage"
+				}
+			},
+			"networkProfile": {
+				"networkInterfaces": [
+					{
+						"id": "[resourceId('Microsoft.Network/networkInterfaces',concat(variables('nicName'), copyindex()))]"
+					}
+				]
+			}
+		}
+	},
+	{
+		"type": "Microsoft.Compute/virtualMachines/extensions",
+		"name": "[concat(variables('vmName'), 'master', '/', variables('extensionName'))]",
+		"apiVersion": "2015-05-01-preview",
+		"location": "[variables('location')]",
+		"dependsOn": [
+			"[concat('Microsoft.Compute/virtualMachines/', variables('vmName'), 'master')]"
+		],
+		"properties": {
+			"publisher": "Microsoft.Azure.Extensions",
+			"type": "DockerExtension",
+			"typeHandlerVersion": "1.0",
+			"settings": {
+				"docker": {
+					"port": "2375"
```

Comment:
This port configuration should not be done because you are not going to expose the docker engine itself to the internet, you'll expose the swarm manager container on the master. So no need to set a port number for the local docker engine here.